spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java

```diff
@@ -90,6 +89,13 @@ public class DataFlowClientAutoConfiguration {
 	@Autowired
 	private @Nullable OAuth2ClientProperties oauth2ClientProperties;
 
+	public DataFlowClientAutoConfiguration() {
```

Comment:
I am not a fan of mixing constructor and field injection. Wdyt about moving it all to CI such as:
```java
	private DataFlowClientProperties properties;

	private RestTemplate restTemplate;

	private ClientRegistrationRepository clientRegistrations;
	
	private OAuth2AccessTokenResponseClient<OAuth2ClientCredentialsGrantRequest> clientCredentialsTokenResponseClient;

	private OAuth2ClientProperties oauth2ClientProperties;

	public DataFlowClientAutoConfiguration(
			DataFlowClientProperties properties,
			@Nullable RestTemplate restTemplate,
			@Nullable ClientRegistrationRepository clientRegistrations,
			@Nullable OAuth2AccessTokenResponseClient<OAuth2ClientCredentialsGrantRequest> clientCredentialsTokenResponseClient,
			@Nullable OAuth2ClientProperties oauth2ClientProperties) {
		this.properties = properties;
		this.restTemplate = restTemplate;
		this.clientRegistrations = clientRegistrations;
		this.clientCredentialsTokenResponseClient = clientCredentialsTokenResponseClient;
		this.oauth2ClientProperties = oauth2ClientProperties;
	}
```

Reply:
+1 for constructor wiring. Anything I can help with here?

Reply:
Thanks @klopfdreh for the offer. We will get this updated and merged shortly though.

Reply:
We are going to accept this proposal as-is for M2 and then revisit in follow on issue w/ the prospect of leveraging the built-in Spring Boot auto-configured `RestTemplateBuilder`.

Reply:
Sounds great! Thanks for all the hard work!